woocommerce/woocommerce-paypal-payments
# Add filters to change allowed ACDC countries

**Issue**: #437

---

### Description

If the ACDC is not supported for a country, our customers needed to manually change the plugin file and add all the missing info for that country.

### Creating Hooks To Allow Filtering Of ACDC Countries

We have discussed and decided to create filters:

- `woocommerce_paypal_payments_supported_country_currency_matrix` - To filter supported countries and their currencies

- `woocommerce_paypal_payments_supported_country_card_matrix` - To filter supported countries and their cards

### How To Use The Filters

Here is the example of how the filters can be used to allow ACDC for Singapore.

```
add_filter('woocommerce_paypal_payments_supported_country_currency_matrix', static function(array $countries): array {

    $countries['SG'] = [
        'AUD',
        'CAD',
        'CHF',
        'CZK',
        'DKK',
        'EUR',
        'GBP',
        'HKD',
        'HUF',
        'JPY',
        'NOK',
        'NZD',
        'PLN',
        'SEK',
        'SGD',
        'USD',
    ];

    return $countries;
});

add_filter('woocommerce_paypal_payments_supported_country_card_matrix', static function(array $countries): array {

    $countries['SG'] = [
        'mastercard' => [],
        'visa'       => [],
        'amex'       => array( 'AUD' ),
    ];

    return $countries;
});
```

### Steps to Test

1. Install and activate the plugin, process onboarding.
2. From Woo settings select the country to be Singapore.
3. Go to PCP settings → PayPal Card Processing Tab and see the message about unsupported country.
4. Put the filters above into theme’s functions.php file
5. Go to PCP settings → PayPal Card Processing Tab.

Now the Country should be supported and you should see an appropriate settings

### Documentation
